### PR TITLE
BSTOR-93: Order total is now affected by articles in the cart and shipping cost

### PR DESCRIPTION
**Pull request by Elrhino**

### CODE REVIEW

File: client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java [hunk 1]
```diff
@@ -30,11 +29,14 @@
 import com.google.web.bindery.event.shared.EventBus;
 
 public class CurrentOrderImpl implements CurrentOrder, HasHandlers {
+    private static final float TAX = 0.14975f;
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java]: Technically, there are two taxes. I'd use two constants instead.

Comment (olafleur) [inline client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java]: Also, we'd have to look what's going on if somebody from the US or another part of the world buy from us.

Comment (olafleur) [inline client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java]: But maybe that's premature for a showcase product. ;)

File: client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/AddressPresenter.java [hunk 2]
```diff
@@ -66,12 +65,13 @@ public void onPaymentDetailsUpdated() {
     private boolean validateContactInfo() {
         ContactInfoDto contactInfo = getContactInfo();
 
-        return !Strings.isNullOrEmpty(contactInfo.getFirstName()) &&
+/*        return !Strings.isNullOrEmpty(contactInfo.getFirstName()) &&
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/AddressPresenter.java]: commented code?

**Comment by olafleur** [comment]
Other than that, LGTM.

File: client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/AddressPresenter.java (hunk 2 above)
Comment (Elrhino) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/AddressPresenter.java]: Yes, I wanted to disabled the contact information validation until development was complete, but I'll just remove the comment for now.

File: client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java [hunk 3]
```diff
@@ -30,11 +29,15 @@
 import com.google.web.bindery.event.shared.EventBus;
 
 public class CurrentOrderImpl implements CurrentOrder, HasHandlers {
+    private static final float TPS = 0.09975f;
+    private static final float TVQ = 0.05f;
```
Comment (meriouma) [inline client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java:33]: If this ever go live, we'll need an other way to calculate the taxes. It depends on the shipping address

File: client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/OrderPresenter.java [hunk 4]
```diff
@@ -27,29 +34,59 @@
...
+
+        void setInternationalPrice(String price);
     }
 
     public static final PermanentSlot<CartItemsPresenter> SLOT_CART_ITEMS = new PermanentSlot<>();
+
     private final CartItemsPresenter cartItemsPresenter;
+    private final CurrencyFormat currencyFormat;
+
+    private CurrentOrder currentOrder;
```
Comment (meriouma) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/OrderPresenter.java:55]: It seems the CurrentOrder can be final

File: client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java (hunk 3 above)
Comment (olafleur) [inline client/src/main/java/com/arcbees/beestore/client/application/CurrentOrderImpl.java:33]: Yes, we already discussed that with @Elrhino . We said that for a showcase that would be fine, but you are right that it'd be more complicated than that for real. ;)

File: client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/PaymentView.ui.xml [hunk 5]
```diff
@@ -24,7 +24,7 @@
 
         <div class="{res.style.paiementMethod}">
             <div class="{comp.style.inputRadio}">
-                <input type="radio" name="paeiment method" id="paypal"/>
+                <input type="radio" name="paeiment method" id="paypal" checked=""/>
```
Comment (meriouma) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/PaymentView.ui.xml:27]: `payment method`? :P

Comment (olafleur) [inline client/src/main/java/com/arcbees/beestore/client/application/widget/sidepanel/checkout/PaymentView.ui.xml:27]: +1

**Pull request merged**